### PR TITLE
feat: Space settings to allow voting with alias

### PR DESCRIPTION
**Pull request by Todmy**
### Issues
#3265 
#3852 

### Changes 
1. Added an option to vote with aliases. To enable aliases, go to Settings -> Voting -> Allow actions with session keys
Screenshot 2023-06-07 at 20 27 14

### How to test
1. Go to the active proposal
2. Create a proposal, delete it (you will be asked to sign your request)
3. Vote in the usual way (to check if it still works)
4. go to Settings -> Voting. And enable `Allow actions with session keys`. Save settings
5. Go back to an active proposal (the previous one or a new active one)
6. Try to vote (the alias voting). In this case, you will be proposed to sign an alias (only once). But next votings will go smoothly without any signing messages
7. Try to create/delete proposal. It should happen without signing message with wallet
8. Go back to Settings -> Voting -> disable `Allow actions with session keys` -> Save settings
9. Go back to an active proposal and try to vote again. This time it will ask you to sign a message for each voting
10. Create/delete a proposal. It should require to sign request

### To-Do
- [ ] 

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [x] I have tested my changes on different screen sizes (sm, md)
- [x] I have tested my changes on a custom domain
- [ ] I have run end-to-end tests `yarn cypress:test:e2e`, and they have passed

### CODE REVIEW

**Comment by samuveth** [comment]
Creating a proposal doesn't work for me after enabling alias. 

I get this error:
image

**Comment by aurelianoB** [comment]
same for me

**Comment by Todmy** [comment]
@samuveth @aurelianoB fixed, please check

**Comment by aurelianoB** [comment]
Thanks @Todmy, both proposal creation and vote with alias works for me now. Let's wait for @samuveth to review anyway

**Comment by 0508511812** [comment]
في خميس، ١٥ يونيو، ٢٠٢٣ في ٤:٥٦ م، كتب aurelianoB ***@***.***

**Comment by 0508511812** [comment]
في خميس، ١٥ يونيو، ٢٠٢٣ في ٦:٣٧ م، كتب WloOo 69 الرياض <
***@***.***>:

**Comment by samuveth** [comment]
Proposal creation is not working for me when I enable alias actions. First I tried without alias setting and it worked, but when I enable it I get:
image

You can test here: https://snapshot-8uiprmz57-snapshot.vercel.app/#/samuv.eth/create
(You will need a Gitcoin Passport attached to the address you use for testing in order to create a proposal)

File: src/composables/useClient.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export function useClient() {
   const { web3 } = useWeb3();
```
Comment (samuveth) [inline src/composables/useClient.ts:10]: This component has become a little hard to read. Can we try something like this?

```js
const ACTIONS = {
  proposal: 'proposal',
  vote: 'vote',
  deleteProposal: 'delete-proposal',
  settings: 'settings',
  deleteSpace: 'delete-space'
};

export function useClient() {

  async function executeAndHandleError(func) {
    isSending.value = true;
    try {
      return await func();
    } catch (e: any) {
      errorNotification(e?.error_description || '');
      return e;
    } finally {
      isSending.value = false;
    }
  }

  function bindWithAlias(actionFn) {
    return (client, payload) => {
      return actionWithAlias(aliasWallet =>
        actionFn(client, payload, {
          provider: aliasWallet,
          address: aliasWallet.address
        })
      );
    };
  }

  async function sendEIP712(space, type, payload) {
    const client = isGnosisSafe.value ? clientGnosisSafe : clientEIP712;
    const shouldUseAlias = space.voting?.aliased && !isGnosisSafe.value;

    switch (type) {
      case ACTIONS.proposal:
        return executeAndHandleError(() => sendProposal(space, payload, client, shouldUseAlias));
      case ACTIONS.vote:
        return executeAndHandleError(() => sendVote(space, payload, client, shouldUseAlias));
      case ACTIONS.deleteProposal:
        return executeAndHandleError(() => sendDeleteProposal(space, payload, client, shouldUseAlias));
      case ACTIONS.settings:
        return executeAndHandleError(() => sendSettings(space, payload, client));
      case ACTIONS.deleteSpace:
        return executeAndHandleError(() => sendDeleteSpace(space, client));
      default:
        throw new Error(`Unknown action type: ${type}`);
    }
  }

  function sendProposal(space, payload, client, shouldUseAlias) {
    const actionFn = shouldUseAlias ? bindWithAlias(proposal) : proposal;

    let plugins = {};
    if (Object.keys(payload.metadata?.plugins).length !== 0)
      plugins = payload.metadata.plugins;

    const proposalPayload = {
      space: space.id,
      type: payload.type,
      title: payload.name,
      body: payload.body,
      discussion: payload.discussion,
      choices: payload.choices,
      start: payload.start,
      end: payload.end,
      snapshot: payload.snapshot,
      plugins: JSON.stringify(plugins),
      app: definedApp,
      from: web3.value.account
    };

    return actionFn(client, proposalPayload);
  }

  // Similarly implement sendVote, sendDeleteProposal, sendSettings, and sendDeleteSpace


}
```

**Pull request closed without merging**